packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -360,9 +360,6 @@ List<Map<String, dynamic>> _extractPlatformMaps(List<Plugin> plugins, String typ
 /// Returns the version of the Android embedding that the current
 /// [project] is using.
 String _getAndroidEmbeddingVersion(FlutterProject project) {
```

Comment:
Unrelated to this PR, just curious - would it be possible / make sense for the logic here and in _getEmbeddingVersion in `platform_plugins.dart` to be the same? (seems like one is deciding based on a manifest key and the other based on Java imports?)

Reply:
We don't have the app's main class in pubspec.yaml, so that would require an extra hop. One possibility is that we get the main app class/package from `AndroidManifest.xml` and then check the import in the java class.  That would still require to parse the XML file.